drivers/docker/driver.go

```diff
@@ -1004,6 +1004,15 @@ func (d *Driver) StopTask(taskID string, timeout time.Duration, signal string) e
 		return fmt.Errorf("failed to parse signal: %v", err)
 	}
 
+	defer func(h *taskHandle) {
```

Comment:
I think a better place to put this would be in the `*taskHandle.run` func. That way it will always happen after we've closed the waitCh (which unblocks WaitTask RPCs). Could you also just make it a func of the taskHandle that way we can just `defer h.stopDockerLogger()` in run.